etcdserver/api/v2store: data race in TestIssue2746WithThree

# version
Git SHA: 49f91d6 (master)
Go Version: go1.14.2
Go OS/Arch: linux/amd64

# issue
https://travis-ci.com/github/etcd-io/etcd/jobs/344148855
https://travis-ci.com/github/etcd-io/etcd/jobs/343926965

## detail:
```
=== RUN   TestIssue2746WithThree
==================
WARNING: DATA RACE
Write at 0x00c007a06f88 by goroutine 764:
  sync/atomic.AddInt64()
      /usr/local/go/src/runtime/race_amd64.s:276 +0xb
  go.etcd.io/etcd/v3/etcdserver/api/v2store.(*Stats).Inc()
      /go/src/go.etcd.io/etcd/etcdserver/api/v2store/stats.go:129 +0x223
  go.etcd.io/etcd/v3/etcdserver/api/v2store.(*store).Get.func1()
      /go/src/go.etcd.io/etcd/etcdserver/api/v2store/store.go:139 +0x76
  go.etcd.io/etcd/v3/etcdserver/api/v2store.(*store).Get()
      /go/src/go.etcd.io/etcd/etcdserver/api/v2store/store.go:149 +0x4ff
  go.etcd.io/etcd/v3/etcdserver.(*reqV2HandlerStore).Get()
      /go/src/go.etcd.io/etcd/etcdserver/v2_server.go:71 +0x2e7
  go.etcd.io/etcd/v3/etcdserver.(*reqV2HandlerEtcdServer).Get()
      <autogenerated>:1 +0xad
  go.etcd.io/etcd/v3/etcdserver.(*RequestV2).Handle()
      /go/src/go.etcd.io/etcd/etcdserver/v2_server.go:155 +0x2c2
  go.etcd.io/etcd/v3/etcdserver.(*EtcdServer).Do()
      /go/src/go.etcd.io/etcd/etcdserver/v2_server.go:131 +0x2b2
  go.etcd.io/etcd/v3/etcdserver/api/v2auth.(*store).requestResource()
      /go/src/go.etcd.io/etcd/etcdserver/api/v2auth/auth_requests.go:132 +0x294
  go.etcd.io/etcd/v3/etcdserver/api/v2auth.(*store).detectAuth()
      /go/src/go.etcd.io/etcd/etcdserver/api/v2auth/auth_requests.go:93 +0x8f
  go.etcd.io/etcd/v3/etcdserver/api/v2auth.(*store).AuthEnabled()
      /go/src/go.etcd.io/etcd/etcdserver/api/v2auth/auth.go:365 +0x38
  go.etcd.io/etcd/v3/etcdserver/api/v2http.hasKeyPrefixAccess()
      /go/src/go.etcd.io/etcd/etcdserver/api/v2http/client_auth.go:127 +0x65
  go.etcd.io/etcd/v3/etcdserver/api/v2http.(*keysHandler).ServeHTTP()
      /go/src/go.etcd.io/etcd/etcdserver/api/v2http/client.go:137 +0x59e
  net/http.(*ServeMux).ServeHTTP()
      /usr/local/go/src/net/http/server.go:2387 +0x288
  go.etcd.io/etcd/v3/etcdserver/api/v2http.requestLogger.func1()
      /go/src/go.etcd.io/etcd/etcdserver/api/v2http/http.go:80 +0x8e
  net/http.HandlerFunc.ServeHTTP()
      /usr/local/go/src/net/http/server.go:2012 +0x51
  net/http.serverHandler.ServeHTTP()
      /usr/local/go/src/net/http/server.go:2807 +0xce
  net/http.(*conn).serve()
      /usr/local/go/src/net/http/server.go:1895 +0x837
Previous read at 0x00c007a06f88 by goroutine 1664:
  go.etcd.io/etcd/v3/etcdserver/api/v2store.(*Stats).clone()
      /go/src/go.etcd.io/etcd/etcdserver/api/v2store/stats.go:89 +0x605
  go.etcd.io/etcd/v3/etcdserver/api/v2store.(*store).Clone()
      /go/src/go.etcd.io/etcd/etcdserver/api/v2store/store.go:764 +0x57c
  go.etcd.io/etcd/v3/etcdserver.(*EtcdServer).snapshot()
      /go/src/go.etcd.io/etcd/etcdserver/server.go:2138 +0xb8
  go.etcd.io/etcd/v3/etcdserver.(*EtcdServer).triggerSnapshot()
      /go/src/go.etcd.io/etcd/etcdserver/server.go:1225 +0x63c
  go.etcd.io/etcd/v3/etcdserver.(*EtcdServer).applyAll()
      /go/src/go.etcd.io/etcd/etcdserver/server.go:1042 +0x200
  go.etcd.io/etcd/v3/etcdserver.(*EtcdServer).run.func8()
      /go/src/go.etcd.io/etcd/etcdserver/server.go:986 +0x53
  go.etcd.io/etcd/v3/pkg/schedule.(*fifo).run()
      /go/src/go.etcd.io/etcd/pkg/schedule/schedule.go:157 +0x11e
Goroutine 764 (running) created at:
  net/http.(*Server).Serve()
      /usr/local/go/src/net/http/server.go:2933 +0x5b6
  net/http/httptest.(*Server).goServe.func1()
      /usr/local/go/src/net/http/httptest/server.go:308 +0xd3
Goroutine 1664 (running) created at:
  go.etcd.io/etcd/v3/pkg/schedule.NewFIFOScheduler()
      /go/src/go.etcd.io/etcd/pkg/schedule/schedule.go:70 +0x2b1
  go.etcd.io/etcd/v3/etcdserver.(*EtcdServer).run()
      /go/src/go.etcd.io/etcd/etcdserver/server.go:871 +0x32c
==================
    TestIssue2746WithThree: testing.go:906: race detected during execution of test
--- FAIL: TestIssue2746WithThree (2.38s)
```

/cc @gyuho @spzala @jingyih @xiang90

--- a/etcdserver/api/v2store/stats.go
+++ b/etcdserver/api/v2store/stats.go
@@ -85,22 +85,22 @@ func newStats() *Stats {
 
 func (s *Stats) clone() *Stats {
 	return &Stats{
-		GetSuccess:              s.GetSuccess,
-		GetFail:                 s.GetFail,
-		SetSuccess:              s.SetSuccess,
-		SetFail:                 s.SetFail,
-		DeleteSuccess:           s.DeleteSuccess,
-		DeleteFail:              s.DeleteFail,
-		UpdateSuccess:           s.UpdateSuccess,
-		UpdateFail:              s.UpdateFail,
-		CreateSuccess:           s.CreateSuccess,
-		CreateFail:              s.CreateFail,
-		CompareAndSwapSuccess:   s.CompareAndSwapSuccess,
-		CompareAndSwapFail:      s.CompareAndSwapFail,
-		CompareAndDeleteSuccess: s.CompareAndDeleteSuccess,
-		CompareAndDeleteFail:    s.CompareAndDeleteFail,
-		ExpireCount:             s.ExpireCount,
-		Watchers:                s.Watchers,
+		GetSuccess:              atomic.LoadUint64(&s.GetSuccess),
+		GetFail:                 atomic.LoadUint64(&s.GetFail),
+		SetSuccess:              atomic.LoadUint64(&s.SetSuccess),
+		SetFail:                 atomic.LoadUint64(&s.SetFail),
+		DeleteSuccess:           atomic.LoadUint64(&s.DeleteSuccess),
+		DeleteFail:              atomic.LoadUint64(&s.DeleteFail),
+		UpdateSuccess:           atomic.LoadUint64(&s.UpdateSuccess),
+		UpdateFail:              atomic.LoadUint64(&s.UpdateFail),
+		CreateSuccess:           atomic.LoadUint64(&s.CreateSuccess),
+		CreateFail:              atomic.LoadUint64(&s.CreateFail),
+		CompareAndSwapSuccess:   atomic.LoadUint64(&s.CompareAndSwapSuccess),
+		CompareAndSwapFail:      atomic.LoadUint64(&s.CompareAndSwapFail),
+		CompareAndDeleteSuccess: atomic.LoadUint64(&s.CompareAndDeleteSuccess),
+		CompareAndDeleteFail:    atomic.LoadUint64(&s.CompareAndDeleteFail),
+		ExpireCount:             atomic.LoadUint64(&s.ExpireCount),
+		Watchers:                atomic.LoadUint64(&s.Watchers),
 	}
 }
 

